src/gtk/srchctrl.cpp

```diff
@@ -0,0 +1,429 @@
+// wxSearchCtrl implementation
+// ============================================================================
+wxBEGIN_EVENT_TABLE(wxSearchCtrl, wxSearchCtrlBase)
+    EVT_CHAR(wxSearchCtrl::OnChar)
+    EVT_TEXT(wxID_ANY, wxSearchCtrl::OnText)
+    EVT_TEXT_ENTER(wxID_ANY, wxSearchCtrl::OnTextEnter)
+wxEND_EVENT_TABLE()
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxSearchCtrl, wxSearchCtrlBase);
+
+// ----------------------------------------------------------------------------
+//  creation/destruction
+// ----------------------------------------------------------------------------
+
+// destruction
+// -----------
+wxSearchCtrl::~wxSearchCtrl()
+{
+    if ( m_entry )
+    {
+        GTKDisconnect(m_entry);
+        g_object_remove_weak_pointer(G_OBJECT(m_entry), (void**)&m_entry);
+    }
+
+#if wxUSE_MENUS
+    delete m_menu;
+#endif // wxUSE_MENUS
+}
+
+// creation
+// --------
+
+void wxSearchCtrl::Init()
+{
+    m_entry = NULL;
+
+#if wxUSE_MENUS
+    m_menu = NULL;
+#endif // wxUSE_MENUS
+
+    m_cancelButtonVisible = false;
+}
+
+bool wxSearchCtrl::Create(wxWindow *parent, wxWindowID id,
+            const wxString& value,
+            const wxPoint& pos,
+            const wxSize& size,
+            long style,
+            const wxValidator& validator,
+            const wxString& name)
+{
+    if ( !PreCreation(parent, pos, size) ||
+         !CreateBase(parent, id, pos, size, (style | wxTE_PROCESS_ENTER),
+                     validator, name) )
+    {
+        wxFAIL_MSG( "wxSearchCtrl creation failed" );
+        return false;
+    }
+
+    GTKCreateSearchEntryWidget();
+
+    if ( HasFlag(wxBORDER_NONE) )
+    {
+        g_object_set (m_widget, "has-frame", FALSE, NULL);
+    }
+
+    GtkEntry * const entry = GetEntry();
+
+    if ( entry )
+    {
+        // Set it up to trigger default item on enter key press
+        gtk_entry_set_activates_default(entry, !HasFlag(wxTE_PROCESS_ENTER));
+
+        gtk_editable_set_editable(GTK_EDITABLE(entry), true);
+#ifdef __WXGTK3__
+        gtk_entry_set_width_chars(entry, 1);
+#endif
+    }
+
+    m_parent->DoAddChild(this);
+
+    if ( entry )
+        m_focusWidget = GTK_WIDGET(entry);
+
+    PostCreation(size);
+
+    if ( entry )
+    {
+        gtk_entry_set_text(entry, wxGTK_CONV(value));
+
+        SetHint(_("Search"));
+
+        GTKConnectChangedSignal();
+        GTKConnectInsertTextSignal(entry);
+        GTKConnectClipboardSignals(GTK_WIDGET(entry));
+    }
+
+    return true;
+}
+
+void wxSearchCtrl::GTKCreateSearchEntryWidget()
+{
+#if wxHAS_GTK_SEARCH_ENTRY
+    if ( wx_is_at_least_gtk3(6) )
+    {
+        m_widget = gtk_search_entry_new();
+    }
+#endif // wxHAS_GTK_SEARCH_ENTRY
+
+    if ( !m_widget )
+    {
+        // No GtkSearchEntry! fallback to the plain GtkEntry.
+        m_widget = gtk_entry_new();
+    }
+
+    g_object_ref(m_widget);
+
+    m_entry = GTK_ENTRY(m_widget);
+    g_object_add_weak_pointer(G_OBJECT(m_entry), (void**)&m_entry);
```

Comment:
Is this really needed here? I'm not sure how exactly can it happen that `m_entry` would be destroyed by anything else other than our own code. There is similar code in `wxComboBox` added by c391cfd6171, but this addresses a special case which is unlikely (impossible?) to happen with this control.

Reply:
Yes your are right, this line should be removed.